docs/CustomMiddlewareChain.md

```diff
@@ -0,0 +1,148 @@
+# Using Custom Middleware Chain
+
+## Middleware
+
+As name suggests it comes in middle of something and that is request and response cycle. It has access to context object which contains request, response objects and any other options that are specific to middleware. Also it has an access to execute next middleware in the chain.
+
+## Step by step procedure
+
+### Implement Middlewares
+
+Create own set of middlewares by implementing [Middleware](../src/IMiddleware.ts) interface. Here two middlewares are created one for handling Logging and another for handling http request and response.
+
+First middleware is passed with the context object containing request, and other middleware specific options. One has to explicitly make call to execute method of the next middleware with context object once the current middleware work is over.
+
+NOTE: Http message handler should set the response object in the context object.
```

Comment:
Move this note about the best practice for http message handlers to the line above the example of the http message handler. 

> **Note**: markdown notes are more notable this way.